tinygrad/tensor.py

```diff
@@ -313,6 +313,8 @@ def num_zeros(step, dim_sz): return 0 if step == 1 or (y := dim_sz % step) == 0
         final_shape.append(1)
     return sliced_tensor.reshape(tuple(final_shape))  # Reshape
 
+  def gather(self, indices, dim): return (self.reshape(*self.shape[:dim+1], *[1]*indices.ndim, *self.shape[dim+1:]).expand(*self.shape[:dim+1], *indices.shape, *self.shape[dim+1:]) * (Tensor.arange(self.shape[dim], dtype=dtypes.int32).reshape(self.shape[dim], *[1]*indices.ndim).expand(self.shape[dim], *indices.shape) == (indices < 0).where(indices+self.shape[dim], indices).unsqueeze(0).expand(self.shape[dim], *indices.shape)).reshape(*[1]*dim, self.shape[dim], *indices.shape, *[1]*(self.ndim-dim-1)).expand(*self.shape[:dim+1], *indices.shape, *self.shape[dim+1:])).sum(dim)
```

Comment:
I think the more readable version of this would be better here. We can sacrifice line count a little bit for readability because this isn't very readable.